src/shogun/mathematics/Math.h

```diff
@@ -924,6 +924,19 @@ class CMath : public CSGObject
 			return ::log(v);
 		}
 
+		/** Computes logarithm of 1 plus argument, v
+		 *
+		 * The method is computed in a way that is accurate
+		 * even if the value of v is near zero
+		 *
+		 * @param v the argument
+		 * @return log base e of (1.0+v) or ln(1.0+v)
+		 */
+		static inline float64_t log1p(float64_t v)
+		{
+			return ::log1p(v);
```

Comment:
@lisitsyn 
Other useful functions can be added in CMath

http://www.johndcook.com/blog/2010/06/07/math-library-functions-that-seem-unnecessary/